metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py

```diff
@@ -349,6 +349,29 @@ def get_views_for_schema(
             )
         return views
 
+    def get_views_for_schema_starts_with(
+        self, schema_name: str, db_name: str
+    ) -> List[SnowflakeView]:
+        views: List[SnowflakeView] = []
+        # Get a grouping of schema names by substring first
+        cur = self.query(SnowflakeQuery.get_views_by_name_substr(schema_name, db_name))
+        for row in cur.fetchall():
```

Comment:
I think it would be better to use `cur. fetchmany` instead of `fetchall`. Then, you don't need all the start_with tricks, and it only requires a minimal code change. -> https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.CursorResult.fetchmany

Would you like to/can you update the PR, or would you prefer us to fix it?

Reply:
Hi Tamas, wouldn't we still have to deal with the fact that Snowflake will fail on schemas with more than 10,000 items?

Reply:
But, if this is really a python related improvement, I think it would be great. I am not sure.

Reply:
Fetchall tries to fetch all the records at once, while fetchmany uses a cursor and paginates over the result. 
I think fetchmany should not be affected by the 10k limit.

Reply:
Okay, that sounds great. I am totally fine if you make the change. Thanks!

Reply:
Sorry, now that I think about it, I know you may be very busy. I can make the change if you want me to.
I could test with the ingestion I am running in my dev environment anyway.  Please let me know.